planetiler-core/src/main/java/com/onthegomap/planetiler/stream/WritableStreamArchive.java

```diff
@@ -0,0 +1,119 @@
+package com.onthegomap.planetiler.stream;
+
+import com.onthegomap.planetiler.archive.WriteableTileArchive;
+import com.onthegomap.planetiler.geo.TileOrder;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.core.util.CloseShieldOutputStream;
+import org.checkerframework.dataflow.qual.Pure;
+
+/**
+ * Base archive for all kinds of simple file streams. This is primarily useful when the file is a named pipe. In that
+ * case data can directly be transformed and consumed by other programs.
+ * <p>
+ * Writing can be parallelized across multiple files (tile_write_threads). For the first file the base path is used. For
+ * consecutive files 1, 2, ... is appended to the base bath.
+ *
+ * <pre>
+ * # create the pipes
+ * mkfifo /tmp/data/output.csv
+ * mkfifo /tmp/data/output.csv1
+ * mkfifo /tmp/data/output.csv2
+ * # start the consumers
+ * consumer_program < /tmp/data/output.csv
+ * consumer_program < /tmp/data/output.csv1
+ * consumer_program < /tmp/data/output.csv2
+ *
+ * # now run planetiler with the options --append --output=/tmp/data/output.csv --tile_write_threads=3
+ * </pre>
+ */
+abstract class WritableStreamArchive implements WriteableTileArchive {
+
+  private final OutputStream primaryOutputStream;
+  private final OutputStreamSupplier outputStreamFactory;
+  @SuppressWarnings("unused")
+  private final StreamArchiveConfig config;
+
+  private final AtomicInteger tileWriterCounter = new AtomicInteger(0);
+
+  private WritableStreamArchive(OutputStreamSupplier outputStreamFactory, StreamArchiveConfig config) {
+    this.outputStreamFactory = outputStreamFactory;
+    this.config = config;
+
+    this.primaryOutputStream = outputStreamFactory.newOutputStream(0);
+  }
+
+  protected WritableStreamArchive(Path p, StreamArchiveConfig config) {
+    this(new FileOutputStreamSupplier(p, config.appendToFile()), config);
+  }
+
+  @Override
+  public final void close() throws IOException {
+    primaryOutputStream.close();
+  }
+
+  @Override
+  public boolean deduplicates() {
+    return false;
+  }
+
+  @Override
+  public TileOrder tileOrder() {
+    return TileOrder.TMS;
+  }
+
+  @Override
+  public final TileWriter newTileWriter() {
+    final int tileWriterIndex = tileWriterCounter.getAndIncrement();
+    if (tileWriterIndex == 0) {
+      return newTileWriter(getPrimaryOutputStream());
+    } else {
+      return newTileWriter(outputStreamFactory.newOutputStream(tileWriterIndex));
+    }
+
+  }
+
+  @Pure
```

Comment:
it's coming with guava
```
[INFO] |  +- com.google.guava:guava:jar:32.1.1-jre:compile
[INFO] |  |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  |  +- org.checkerframework:checker-qual:jar:3.33.0:compile
[INFO] |  |  +- com.google.errorprone:error_prone_annotations:jar:2.18.0:compile
[INFO] |  |  \- com.google.j2objc:j2objc-annotations:jar:2.8:compile
```
and https://checkerframework.org/api/org/checkerframework/dataflow/qual/Pure.html

I was initially using the method from the constructor... but since I changed that, I'll remove the annotation now